pkg/storage/gcp/gcp.go

```diff
@@ -0,0 +1,63 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"cloud.google.com/go/storage"
+	"github.com/gomods/athens/pkg/config/env"
+	"google.golang.org/api/option"
+)
+
+// Storage implements the (./pkg/storage).Backend interface
+type Storage struct {
+	bucket       *storage.BucketHandle
+	baseURI      *url.URL
+	closeStorage func() error
+}
+
+// NewWithCredentials returns a new Storage instance authenticated using the provided
+// ClientOptions. The bucket name to be used will be loaded from the
+// environment variable ATHENS_STORAGE_GCP_BUCKET.
+//
+// The ClientOptions should provide permissions sufficient to read, write and
+// delete objects in google cloud storage for your project.
+func NewWithCredentials(ctx context.Context, cred option.ClientOption) (*Storage, error) {
+	storage, err := storage.NewClient(ctx, cred)
+	if err != nil {
+		return nil, fmt.Errorf("could not create new storage client: %s", err)
+	}
+	bucketname, err := env.GcpBucketName()
+	if err != nil {
+		return nil, err
+	}
+	u, err := url.Parse(fmt.Sprintf("https://storage.googleapis.com/%s", bucketname))
```

Comment:
can you please use this on top of URI before setting baseURI
https://github.com/gomods/athens/blob/46699719a030675d6a0006d5bfcf176aef515fdf/pkg/config/env/cdn.go#L11

Reply:
ok i see it in base uri, Ignore me